### DISCUSSION TITLE
请问下，怎么原样导出导入所有的api

### DISCUSSION THREAD

**Discussion by bewareilive** [Postcatlab/postcat#96] [Q&A]
使用导出openapi会丢失很多信息。想使用导出做数据备份。

还有就是插件经常下载不下来

**Comment by kungfuboy** [comment]
@bewareilive 有可能是我们开发时使用的测试用例过于简单，我们会逐步完善对 openapi 的解析。
插件下载的问题，我们会尽快解决，谢谢！

**Reply by scarqin** [reply]
@bewareilive 具体是指哪个插件呢？

**Reply by bewareilive** [reply]
> @bewareilive 具体是指哪个插件呢？

我下的是postman的导入的那个。
应该是网络问题，插件是在github吗？我这GitHub经常会连不上。
那个也没有进度条之类的指示，不知道是下载慢还是压根连不上，放一个小时也下不下来，然后所有按钮点不了，只能关进程了。

打算等这个稳定了替换postman。
因为调试文件要和项目一起存档，或者发送给其他同事使用（内网）。
所以离线情况下的导入导出，是我对着类软件的第一考虑，目前市面上的几乎试了个遍，暂时还是只能使用postman。

非常期待这个软件的后续

**Reply by scarqin** [reply]
@bewareilive 了解，那个插件暂时下架了，npm 上传包出了些问题，感谢反馈，我们会在解决后再次上架。

了解，团队里面不同人使用不同的工具，所以用离线文件打通/分享是重点功能，我们还在继续完善，做插件市场也是希望以后可以通过插件让使用者能更方便地共享数据，感谢支持～

**Comment by bewareilive** [comment]
每隔几天看下，等出导出了再用

**Reply by scarqin** [reply]
@bewareilive Postman 导入的在做，导出插件我们更新了几个版本了，请提供一下具体缺失了以便于我们定位到问题尽快修复～

**Reply by bewareilive** [reply]
> @bewareilive Postman 导入的在做，导出插件我们更新了几个版本了，请提供一下具体缺失了以便于我们定位到问题尽快修复～

![image](https://user-images.githubusercontent.com/26020170/185020912-613888bd-2231-42b6-adc3-31682160b57c.png)

刚尝试了下，点击导出页面无反应，控制台报错。

**Reply by bewareilive** [reply]
使用的1.5.1版本

**Reply by scarqin** [reply]
@bewareilive 插件的版本是？尝试下卸载了重装看看，看起来是依赖问题

**Reply by bewareilive** [reply]
> @bewareilive 插件的版本是？尝试下卸载了重装看看，看起来是依赖问题

在两台装了，因为其中一台是新装的，所以肯定不是老版本问题。
仔细检查后发现：
需要右键使用管理员打开才可以（可能是因为我以管理员安装的？）。

导出后看了下，问题还是很多：
1、我建了几个目录，还有在根目录的请求，但只导出了其中一个目录的。
2、我希望达到的目的是，我导出的文件，别人导入后，和我本机上看到的完全相同，而现在的并不是的。
      这一点，感觉类似[mockoon](https://mockoon.com/)。他里面有个导出openapi的功能，但我们从来不用。虽然导出的opanapi可以在其他软件或自己导入，却会缺失很多东西。
      通常都是直接用他的工程文件json。这个拿过去别人打开，就是完全一致的。
3、另外，有时并不想将所有东西都给别人，比如只选择导出某个目录下的。

**Reply by scarqin** [reply]
@bewareilive 你用导出的应用场景主要是和其他人配合吗？

因为涉及到数据转换， OpenAPI 和平台的字段不一定完全匹配的，没法实现完全双向无失真的同步。

**Reply by bewareilive** [reply]
@scarqin 对的，上面其实就说过，_“因为调试文件要和项目一起存档，或者发送给其他同事使用（内网）。”_
主要应用场景就是 api调试，接口都是只能在客户的内网环境调试，调试完把调试的接口和一些返回数据导出来，带到开发环境使用。
或者是反过来，开发环境保存的接口，在客户的内网直接调试。

这种需求下，目前只能使用postman，另几个国产的api调试软件，都必须连接互联网无法使用。

**Reply by scarqin** [reply]
@bewareilive 这周末我开发俩插件上架。

**Reply by bewareilive** [reply]
@scarqin 非常期待，下周试试。一直关注更新的，但内容里没看到自己需要的功能

**Reply by scarqin** [reply]
@bewareilive 哈哈哈♥️

**Reply by scarqin** [reply]
@bewareilive 发布了～
可以直接在线体验一下：
https://www.eoapi.io/zh/home/extension/list

<img width="1493" alt="image" src="https://user-images.githubusercontent.com/15225835/204100205-c0f953d3-892e-499d-903b-913cff977eab.png">

**Reply by scarqin** [reply]
https://docs.eoapi.io/docs/local-install-extension.html
如果要离线安装插件，教程在这里。

**Reply by bewareilive** [reply]
@scarqin 给个小建议，将eolink和eoapi插件的图片稍微区别，或者起码在导入导出界面，带上文字。
早上导出好着，导入试了半天不成功，也没有提示。
打算卸载掉插件重装试试。然后才发现装错成eolink了，一直在那个界面尝试🤣

**Reply by scarqin** [reply]
@bewareilive 哈哈哈是的，开发时就注意到了，但是还没空设计，见谅见谅
